### ISSUE TITLE
Add user defined patterns

### ISSUE THREAD

**Issue by gibberish-reprimand** [intiface/intiface-central#171]
I would like to request the feature that allows users to define named patterns/actions that interface central clients can turn on and off. This will improve accessibility for users that have requirements on how devices should be used, as the user can set up what the device activation means.

I'm imagining the user can set up a name, and per device and feature patterns within the interface central app. These named actions can then be queried by clients connecting to interface central and they can then expose in their settings menu what to use the patterns for, think setting up key bindings for a game.

Proposed Implementation:

Pattern Definition Menu:
Integrate a new menu within the existing settings interface where users can create and name custom vibration patterns.

Allow users to setup a pattern by filling out a list of actions that define a pattern. Where an action would be a device, feature, duration, and sequence. Where sequence is a graph or similar input ui that can define over time patten that device should follow.

API Exposure:
Extend the interface central API to include a method for sending these custom patterns to devices.

I'm imagining this would be the rough client interface psudocode:

interfacecentral.get_patterns()

Parameters:

Behavior:
Returns an Array[String] that lists all patterns that have been set up in the settings.

interfacecentral.get_active_patterns()

Parameters:

Behavior:
Returns an Array[String] that describe the patterns currently being run

interfacecentral.stop_active_pattern("pattern_name")

Parameters:

pattern_name (String): The name of the predefined pattern.

Behavior:
Stops the pattern if it's currently running, I imagine having something like pause would also be a useful option.

interfacecentral.send_pattern("pattern_name", loop=true, intensity=1)

Parameters:

pattern_name (String): The name of the predefined pattern.

loop (Boolean, default=true): Determines whether the pattern should loop continuously.

intensity (Float, default=1.): 0.-1. float that is used by interfacecentral to adjust the pattern intensity, where 0 will not do anything and 1 will play the pattern at the full intensity the pattern is defined as in the settings menu.
This means if the user wants to set a maximum it can be done and it is abstracted away from the interface central client implications. 

Behavior:

If loop is set to true, the specified pattern will restart when completed and play continuously until stopped or the client is disconnected.

If loop is set to false, the specified pattern will play once, and then the previous pattern will resume.

Notes:
Since you can have multiple patterns and they can use the same device or device feature in a pattern I imagine there should be a defined method to describe tie breaker to determine what wins when two patterns want to use the same device feature.
But that probably just means the order they are listed/defined in.

Benefits:

Enhances user experience and accessibility by providing customizable and reusable device feature patterns.

Simplifies the process of assigning complex patterns to device functions and allows for simplified combinations of multiple devices or device features.

Increases accessibility in device control through the interface central. Making it similar to an input system for a game where the action is defined so the code only knows about the "buzz" input happened whilst "buzz" is defined in interface-central settings.

**Comment by qdot** [comment]
Closing this because it's more of a buttplug issue than intiface (and also this completely misunderstands how our system works at pretty much all levels). 

We're discussing similar but more in line with how our system works at https://github.com/buttplugio/buttplug/issues/810
